File: app-sdk/python/iagent_sdk/iagent/model/resource.py
# -*- coding: utf-8 -*-

# Copyright (C) 2017 Intel Corporation.  All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

class Resource(object):
    def equals(self, obj):
        return self.href == obj.href
    def get_absolute_uri(self):
        if self.device is not None:
		if self.href.startswith("/"):
			return self.device.get_absolute_uri() + self.href
		else:
			return self.device.get_absolute_uri() + "/" + self.href
	else:
		return self.href
    def get_attrs(self):
        return self.attrs
    def get_current_value(self, property_name):
        return None
    def get_groups(self):
        return self.groups
    def get_href(self):
        return self.href
    def get_owner_device(self):
        return self.device
    def get_owner_groups(self):
        return self.groups
    def get_properties_config(self, property_key):
        return self.__property_config[property_key]
    def get_resource_type(self):
        return self.resource_type
    def hash_code(self):
        return hash(self.href)
    def __init__(self, resource=None):
        self.href = None
	self.resource_type = []
	self.attrs = None
	self.device = None
	self.groups = []
	self.__property_config = {}
	self.__property_data_cache = {}
	if resource is not None:
		self.href = resource.href
		self.resource_type = resource.resource_type
		self.attrs = resource.attrs
		self.device = resource.device
		self.__property_config = resource.__property_config
		self.__property_data_cache = resource.__property_data_cache
		self.groups = resource.groups
	

File: app-sdk/java/code/src/com/intel/idrml/iagent/model/Resource.java
/*
 * Copyright (C) 2017 Intel Corporation.  All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.idrml.iagent.model;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.alibaba.fastjson.JSONObject;

public class Resource {
    //@SerializedName("href")
    public String href; // uri, defined as href in interface

    public List<String> resourceType;

    //@SerializedName("attrs")
    public String attrs; // configured in cloud

    //@Expose(serialize = false, deserialize = false)
    public DeviceInfo device;

    //@SerializedName("groups")
    public List<String> groups;

    //@Expose(serialize = false, deserialize = false)
    private Map<String, String> propertyConfig;

    //@Expose(serialize = false, deserialize = false)
    private Map<String, String> propertyDataCache; //TODO DataCach shall be defined

    public Resource() {

    }

    public Resource(Resource resource) {
        this.href = resource.href;
        this.resourceType = resource.resourceType;
        this.attrs = resource.attrs;
        this.device = resource.device;
        this.propertyConfig = resource.propertyConfig;
        this.propertyDataCache = resource.propertyDataCache;
        this.groups = resource.groups;
    }

    public String getHref() {
        return href;
    }

    public String getAttrs() {
        return attrs;
    }

    public List<String> getGroups() {
        return groups;
    }

    public String getPropertiesConfig(String propertyKey) {
        return propertyConfig.get(propertyKey);
    }

    //TODO
    public String getCurrentValue(String propertyName) {
        return null;
    }

    public String getAbsoluteUri() {
        if (device != null) {
            return device.getAbsoluteUri() + ((href.startsWith("/"))?(href):("/" + href));
        } else {
            return href;
        }
    }

    public List<String> getResourceType() {
        return resourceType;
    }

    public DeviceInfo getOwnerDevice() {
        return device;
    }

    public List<String> getOwnerGroups() {
        return groups;
    }

    @Override
    public boolean equals(Object obj) {
        return href.equals(((Resource) obj).href);
    }

    @Override
    public int hashCode() {
        return href.hashCode();
    }

}
